windows-server-container-tools/CleanupContainerHostNetworking/WindowsContainerNetworking-LoggingAndCleanupAide.ps1

```diff
@@ -0,0 +1,445 @@
+        }
+    }
+    finally
+    {
+        if ($hns.Status -eq "Running")
+        {
+            net start hns
+        }
+    }
+
+    ipconfig /allcompartments /all > $LogsPath"\ipconfig.txt"
+
+    if (Test-Path "C:\Windows\System32\vfpctrl.exe")
+    {
+        $allSwitches = Get-VMSwitch
+        foreach ($switch in $allSwitches)
+        {
+            GetAllPolicies -switchName $switch.Name > $LogsPath"\SwitchPolicies_$($switch.Name).txt"
+        }
+
+        Get-Service vfpext | FL * > $LogsPath"\vfpext.txt"
+    }
+
+
+    # Get docker version
+    docker -v > $LogsPath"\docker_v.txt"
+
+    # Get docker info
+    docker info > $LogsPath"\docker_info.txt"
+
+    # Get list containing IDs of current container networks on host
+    $networks=docker network ls -q
+    # Initialize file for recording network info
+    "Number of container networks currently on this host: "+$networks.Length+"`\n" > $LogsPath"\docker_network_inspect.txt"
+    # Inspect each network
+    foreach($network in $networks)
+    {
+      $addMe = docker network inspect $network
+      Add-Content $LogsPath"\docker_network_inspect.txt" $addMe
+    }
+
+    Get-VMSwitch | FL * > $LogsPath"\GetVMSwitch.txt"
+    Get-NetNat | FL * > $LogsPath"\GetNetNat.txt"
+    Get-NetNatStaticMapping | FL *> $LogsPath"\GetNetNatStaticMapping.txt"
+    Get-Service winnat | FL * > $LogsPath"\WinNat.txt"
+    Get-Service mpssvc | FL * > $LogsPath"\mpssvc.txt"
+    Get-NetAdapter -IncludeHidden | FL * > $LogsPath"\GetNetAdapters.txt"
+    Get-NetAdapterBinding -IncludeHidden | FL * > $LogsPath"\GetNetAdapterBinding.txt"
+
+    Get-ChildItem HKLM:\SYSTEM\CurrentControlSet\Services\hns -Recurse > $LogsPath"\HNSRegistry.txt"
+    Get-ChildItem HKLM:\SYSTEM\CurrentControlSet\Services\vmsmp -Recurse > $LogsPath"\vmsmpRegistry.txt"
+}
+
+# -----------------------------------------------------------------------
+# This function generates a string of 8 random characters
+# Parameters: _
+# Returns: 8 character string
+# -----------------------------------------------------------------------
+function GenerateRandomSuffix
+{
+    $rand = New-Object System.Random
+    $len = 8
+    [string]$namingSuffix = ''
+    for($i = 0; $i -lt $len; $i++)
+    {
+        $namingSuffix += [char]$rand.Next(65,90)
+    }
+
+    return $namingSuffix
+}
+
+
+# -----------------------------------------------------------------------
+# SCRIPT STARTS HERE
+# -----------------------------------------------------------------------
+try
+{   
+    # FIRST, WHICH VERSION OF DOCKER IS RUNNING ON THIS HOST?
+    # *******************************************************
+
+	Try
+	{
+		# Docker EE
+	    Get-Service docker -ErrorAction silentlycontinue
+	}
+	Catch [Microsoft.PowerShell.Commands.ServiceCommandException]
+	{
+		# Docker CE
+	    Get-Service com.docker.service -ErrorAction stop
+	}
+	Catch
+	{
+		write-host "ERROR: Docker is not running on this host. Please start Docker and try again." -ForegroundColor Red
```

Comment:
do we still see any cases where the Docker service won't start due to networking problems? I know that was a common case due to HNS bugs in tech preview